server/protocols/webadmin/webadmin-cassandra/src/main/java/org/apache/james/webadmin/routes/CassandraMailboxMergingRoutes.java

```diff
@@ -0,0 +1,102 @@
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.routes;
+
+import javax.inject.Inject;
+
+import org.apache.james.mailbox.cassandra.ids.CassandraId;
+import org.apache.james.mailbox.cassandra.mail.task.MailboxMergingTask;
+import org.apache.james.mailbox.cassandra.mail.task.MailboxMergingTaskRunner;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.webadmin.Constants;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.dto.MailboxMergingRequest;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.apache.james.webadmin.utils.ErrorResponder.ErrorType;
+import org.apache.james.webadmin.utils.JsonExtractException;
+import org.apache.james.webadmin.utils.JsonExtractor;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+public class CassandraMailboxMergingRoutes implements Routes {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraMailboxMergingRoutes.class);
+
+    public static final String BASE = "/cassandra/mailbox/merging";
+
+    private final MailboxMergingTaskRunner mailboxMergingTaskRunner;
+    private final CassandraId.Factory mailboxIdFactory;
+    private final JsonExtractor<MailboxMergingRequest> jsonExtractor;
+
+    public static String PARTIAL_MERGING_PROCESS = "An error lead to partial merging process.Check server logs";
+
+    @Inject
+    public CassandraMailboxMergingRoutes(MailboxMergingTaskRunner mailboxMergingTaskRunner, CassandraId.Factory mailboxIdFactory) {
+        this.mailboxMergingTaskRunner = mailboxMergingTaskRunner;
+        this.mailboxIdFactory = mailboxIdFactory;
+        this.jsonExtractor = new JsonExtractor<>(MailboxMergingRequest.class);
+    }
+
+    @Override
+    public void define(Service service) {
+        service.post(BASE, this::mergeMailboxes);
+    }
+
+    private Object mergeMailboxes(Request request, Response response) {
+        try {
+            LOGGER.debug("Cassandra upgrade launched");
+            MailboxMergingRequest mailboxMergingRequest = jsonExtractor.parse(request.body());
+            CassandraId originId = mailboxIdFactory.fromString(mailboxMergingRequest.getMergeOrigin());
+            CassandraId destinationId = mailboxIdFactory.fromString(mailboxMergingRequest.getMergeDestination());
+
+            new MailboxMergingTask(mailboxMergingTaskRunner, originId, destinationId)
+                .run()
+                .ifCompleted(() -> response.status(HttpStatus.NO_CONTENT_204))
```

Comment:
Aren't we affraid that the merging take too long and that the http request time out before the result?

Reply:
Did not see that after hours long migrations.

There is no reverse proxy in the equation so no timeout.

Reply:
still it's a bad design. should return 201 Created and then should be monitorable.